commands/chargen.py

```diff
@@ -0,0 +1,190 @@
+    key = "rem_charcreate_cmdset"
+    priority = 2
+    mergetype = "Remove"
+
+    def at_cmdset_creation(self):
+        """Populate CmdSet"""
+        self.add(CmdCharCreate())
+
+
+class ChargenICCmdSet(CmdSet):
+    """Command set to override the @ic command on Player"""
+    key = 'chargen_ic_cmdset'
+    priority = 2
+
+    def at_cmdset_creation(self):
+        self.add(CmdIC())
+
+
+class CmdIC(default_cmds.CmdIC):
+    """
+    control an object you have permission to puppet
+
+    Usage:
+      @ic <character>
+
+    Go in-character (IC) as a given Character.
+
+    This will attempt to "become" a different object assuming you have
+    the right to do so. Note that it's the PLAYER character that puppets
+    characters/objects and which needs to have the correct permission!
+
+    You cannot become an object that is already controlled by another
+    player. In principle <character> can be any in-game object as long
+    as you the player have access right to puppet it.
+    """
+    def func(self):
+        """Don't allow puppeting unless the chargen_complete attribute is set."""
+        if self.args:
+            new_character = search.object_search(self.args)
+            if new_character:
+                new_character = new_character[0]
+                if not new_character.db.chargen_complete:
+                    self.session.execute_cmd('@charcreate {}'.format(
+                        new_character.key
+                    ))
+                    return
+            else:
+                self.msg("That is not a valid character choice.")
+                return
+        super(CmdIC, self).func()
+
+
+class CmdCharCreate(MuxPlayerCommand):
+    """
+    create a new character
+
+    Usage:
+      @charcreate <charname>
+
+    Create a new character. You may use upper-case letters in the
+    name - you will nevertheless always be able to access your
+    character using lower-case letters if you want.
+    """
+    key = "@charcreate"
+    locks = "cmd:pperm(Players)"
+    help_category = "General"
+
+    def func(self):
+        "create the new character"
+        player = self.player
+        session = self.session
+        if not self.args:
+            self.msg("Usage: @charcreate <charname>")
+            return
+        key = self.args.strip()
+
+        charmax = _MAX_NR_CHARACTERS if _MULTISESSION_MODE > 1 else 1
+
+        if not player.is_superuser and \
+            (player.db._playable_characters and
+                len(player.db._playable_characters) >= charmax):
+            self.msg("You may only create a maximum of %i characters." % charmax)
+            return
+
+        # create the character
+        from evennia.objects.models import ObjectDB
+
+        start_location = ObjectDB.objects.get_id(settings.START_LOCATION)
+        default_home = ObjectDB.objects.get_id(settings.DEFAULT_HOME)
+        typeclass = settings.BASE_CHARACTER_TYPECLASS
+        permissions = settings.PERMISSION_PLAYER_DEFAULT
```

Comment:
Good point. I can check the typeclass in  `@ic` and just pass any non-character object through to the default `@ic`.